src/TurboMqtt/IO/Tcp/FakeMqttTcpServer.cs

```diff
@@ -190,8 +193,9 @@ private async Task ProcessClientAsync(Socket socket)
                 }
             }
 
-            // send a disconnect message
-            handle.DisconnectFromServer();
+            if(!closed)
+                // send a disconnect message
+                handle.DisconnectFromServer();
```

Comment:
Clean up fake server code, if the socket's already closed, no need to send a disconnect packet